### PR TITLE
node 20 support

### PR DESCRIPTION
**Pull request by shakyShane**
https://app.asana.com/0/1201614831475344/1206978084282836/f

### TL;DR

This PR upgrades the Node.js version used in the GitHub actions workflows from 18 to 20. It also adjusts some test cases.

### What changed?

- The Node.js version in the GitHub actions workflows and the .nvmrc file was updated from 18.x to 20.x.
- Some tests updated to handle the random IDs we get now because of the newer node version

### How to test?

After merging, confirm that the GitHub actions workflows run successfully using the new Node.js version.

### Why make this change?

Screenshot 2024-04-04 at 10 48 47 AM

---

### CODE REVIEW

**Comment by shakyShane** [comment]
* **#948** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/948?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shakyShane and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/messaging/index.js [hunk 1]
```diff
@@ -36,12 +36,22 @@ export class MessagingContext {
...
         this.featureName = params.featureName
         this.env = params.env
+
+        /**
+         * This allows for deterministic tests in all environments. For example,
+         * in a test situation you can ensure the ID is known.
+         */
+        this.createId = params.createId || ((name) => {
+            return globalThis?.crypto?.randomUUID?.() || name + '.response'
+        })
```
Comment (shakyShane) [inline packages/messaging/index.js:54]: This behaves exactly as before, but also now acts as DI for overriding the side-effecting nature of `globalThis?.crypto?.randomUUID`

File: packages/messaging/index.js [hunk 2]
```diff
@@ -97,10 +97,14 @@ export class Messaging {
...
+     * @param {object} [overrides]
+     * @param {string} [overrides.id] - the ID sent with outgoing requests
      * @return {Promise<any>}
      */
-    request (name, data = {}) {
-        const id = globalThis?.crypto?.randomUUID?.() || name + '.response'
+    request (name, data = {}, options = {}) {
+        // first try the override, then global crypto, falling back to <name>.response
+        const id = options.id || globalThis?.crypto?.randomUUID?.() || name + '.response'
+
```
Comment (shakyShane) [inline packages/messaging/index.js:107]: added this 3rd param as an object onto which we can add runtime overrides. Right now just accepting an `id`, but can also be used later to send things like AbortControllers through too.

File: packages/messaging/index.js [hunk 3]
```diff
@@ -97,10 +97,14 @@ export class Messaging {
      *
      * @param {string} name
      * @param {Record<string, any>} [data]
+     * @param {object} [overrides]
+     * @param {string} [overrides.id] - the ID sent with outgoing requests
      * @return {Promise<any>}
      */
-    request (name, data = {}) {
-        const id = globalThis?.crypto?.randomUUID?.() || name + '.response'
+    request (name, data = {}, options = {}) {
```
Comment (jonathanKingston) [inline packages/messaging/index.js:104]: Not sure why TS isn't whining about this:

```suggestion
    request (name, data = {}, overrides = {}) {
```

**Review by jonathanKingston — APPROVED** [review]

**Pull request merged**